Add basic test for curl_unescape

eduardogonrod opened the pull request.

tpunt commented on ext/curl/tests/curl_unescape_basic001.phpt:
```diff
@@ -0,0 +1,13 @@
+--TEST--
+curl_unescape basic test
+--SKIPIF--
+<?php if(!extension_loaded('curl')) print 'skip'; ?>
```
The `curl_escape` and `curl_unescape` functions aren't necessarily going to be loaded (see [this related test for an example](https://github.com/php/php-src/blob/5f4a417537117ed5baadb4c0a1c738cdb4638c86/ext/curl/tests/curl_escape.phpt)).

cmb69 replied:
Also please provide a reason why the test has been skipped, such as "cURL extension not available".

tpunt commented on ext/curl/tests/curl_unescape_basic001.phpt:
```diff
@@ -0,0 +1,13 @@
+var_dump(curl_unescape($ch, '%20%21%22%23%24%25%26%27%28%29%2A%2B%2C%2D%2E%2F%30%31%32%33%34%35%36%37%38%39%3A%3B%3C%3D%3E%3F%40%41%42%43%44%45%46%47%48%49%4A%4B%4C%4D%4E%4F%50%51%52%53%54%55%56%57%58%59%5A%5B%5D%5E%5F%60%61%62%63%64%65%66%67%68%69%6A%6B%6C%6D%6E%6F%70%71%72%73%74%75%76%77%78%79%7A%7B%7C%7D%7E'));
+
+?>
+--EXPECTF--
```
`--EXPECTF--` is for formatted result matching. It looks like you should simply be using `--EXPECT--`.

eduardogonrod commented:
I didn't realise that there is another test for this. I checked http://gcov.php.net/ and thought that function was not covered, that was the reason I provided a test for this function. Should I close this then? However I have updated the skip comment and changed to EXPECT instead of EXPECTF.

nikic commented:
Looks like the test only works with libcurl >= 7.21.2 and gcov uses an older version. As the function is already tested (and the test just happens not to run on gcov), I'm closing this PR.
